### PR TITLE
Add Signature and Signature Verification support

### PR DESCRIPTION
**Pull request by dbluhm**
This PR updates the encryption envelope library to the most recent version which includes helpers for dealing with signed attachments in DIDComm messages.

In the toolbox, as developer oriented piece of software, signatures on messages sent during connection establishment are currently not being properly verified. As we move into OOB/DID Exchange support, we will be required to both verify signatures as well as provide signatures on our DID Exchange request. This PR adds the necessary building blocks to make this possible.

### CODE REVIEW

File: src/renderer/components/ConnectionsProtocol.js
Comment (dbluhm) [inline src/renderer/components/ConnectionsProtocol.js:76]: These lines are of particular note as they demonstrate how the newly added methods are to be used. Observe the output in your console on newly created connections with the toolbox.

**Comment by dbluhm** [comment]
@mepeltier @PeterStrob please review.

**Comment by dbluhm** [comment]
After a review from @mepeltier and @PeterStrob, @frostyfrog please give your approval as a committer :slightly_smiling_face:

**Comment by frostyfrog** [comment]
Will do. Am I understanding correctly that you would like me to wait for their review?

**Review by mepeltier — APPROVED** [review]
It all looks good to me. Seems like a pretty straightforward way to use and verify signatures

**Comment by PeterStrob** [comment]
Don't think I've been added as a reviewer here but it looks good!

**Comment by dbluhm** [comment]
Thanks for the reviews @mepeltier and @PeterStrob! @frostyfrog this should be good for your approval now :slightly_smiling_face:

**Review by frostyfrog — APPROVED** [review]
These changes look simple enough to me, and I'm satisfied with the reviews from the others. Marking this as approved and getting it merged in. :)

**Pull request merged**
